src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs

```diff
@@ -100,14 +113,33 @@ internal LoadedTypeItem CreateLoadedTypeItem(NodeSearchElement element)
                         : string.Empty
             };
 
+            InitializeLoadedTypeItem(element, item);
+            return item;
+        }
+
+        /// <summary>
+        /// Initialize a LoadedTypeItem object
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="item"></param>
+        protected void InitializeLoadedTypeItem(NodeSearchElement element, LoadedTypeItem item)
```

Comment:
This is the one that after an instance of LoadedTypeItem or its derived class is created, the data need to be initialized to set iconUrl or update fully qualified name. This will be the same in the base and the derived classes.